### PR TITLE
feat: add google cloud run starter workflows

### PR DESCRIPTION
**Pull request by verbanicm**
Added new starter workflows for Google Cloud Run

### CODE REVIEW

File: deployments/properties/google-cloudrun-docker.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Build and Deploy to Cloud Run",
+  "description": "Build a Docker container, publish it to Google Artifact Registry, and deploy to Google Cloud Run.",
+  "creator": "Google Cloud",
+  "iconName": "google-cloud",
+  "categories": ["Deployment", "Containers", "Cloud Run", "Serverless"]
```
Comment (bharathkkb) [inline deployments/properties/google-cloudrun-docker.json:6]:
```suggestion
  "categories": ["Deployment", "Containers", "Dockerfile", "Cloud Run", "Serverless"]
```

File: deployments/google-cloudrun-docker.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+#   3. Build a docker container
+#   4. Publish it to Google Artifact Registry
+#   5. Deploy it to Cloud Run
+#
+# To configure this workflow:
+#
+# 1. Ensure the required Google Cloud APIs are enabled:
+#
+#    Cloud Run            run.googleapis.com
+#    Artifact Registry    artifactregistry.googleapis.com)
```
Comment (bharathkkb) [inline deployments/google-cloudrun-docker.yml:14]:
```suggestion
#    Artifact Registry    artifactregistry.googleapis.com
```

File: deployments/google-cloudrun-docker.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# This workflow run the following steps when there is a push to the $default-branch branch:
```
Comment (bharathkkb) [inline deployments/google-cloudrun-docker.yml:1]:
```suggestion
# This workflow will build and push a Docker container to Google Artifact Registry and deploy it on Cloud Run when a commit is push to the $default-branch branch:
```

File: deployments/google-cloudrun-docker.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      # - name: Google Auth
+      #   id: auth
+      #   uses: "google-github-actions/auth@v0"
+      #   with:
+      #     credentials_json: "${{ secrets.GCP_CREDENTIALS }}"
+
+      # NOTE: If you already have a container image, these Docker steps can be omitted
+
+      # BEGIN - Docker auth and build
+      #
```
Comment (bharathkkb) [inline deployments/google-cloudrun-docker.yml:75]: I dont think we need this?

File: deployments/google-cloudrun-source.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# This workflow run the following steps when there is a push to the $default-branch branch:
```
Comment (bharathkkb) [inline deployments/google-cloudrun-source.yml:1]:
```suggestion
# This workflow will deploy source code on Cloud Run when a commit is push to the $default-branch branch:
```

**Review by bharathkkb — APPROVED** [review]
Some minor nits but LGTM!

File: deployments/google-cloudrun-docker.yml (hunk 2 above)
Comment (sethvargo) [inline deployments/google-cloudrun-docker.yml:14]:
```suggestion
#    Artifact Registry    artifactregistry.googleapis.com
```

File: deployments/google-cloudrun-docker.yml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+#   5. Deploy it to Cloud Run
+#
+# To configure this workflow:
+#
+# 1. Ensure the required Google Cloud APIs are enabled:
+#
+#    Cloud Run            run.googleapis.com
+#    Artifact Registry    artifactregistry.googleapis.com)
+#
+# 2. Create and configure a Workload Identity Provider for GitHub (https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)
```
Comment (sethvargo) [inline deployments/google-cloudrun-docker.yml:16]:
```suggestion
# 2. Create and configure a Workload Identity Federation for GitHub (https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)
```

File: deployments/google-cloudrun-docker.yml [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+#
+# 3. Ensure the required IAM permissions are granted:
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply deployments/google-cloudrun-docker.yml:18]: Agreed - let's list the IAM roles for Cloud Run and AR, but add the standard disclaimer about Principal of Least Privilege.

File: deployments/google-cloudrun-docker.yml [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+name: Build and Deploy to Cloud Run
+
+on:
+  push:
+    branches:
+      - $default-branch
+
+env:
+  PROJECT_ID: ${{ secrets.GCP_PROJECT }}
```
Comment (sethvargo) [inline deployments/google-cloudrun-docker.yml:40]: I wouldn't consider this to be a secret.

File: deployments/google-cloudrun-docker.yml [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  GAR_LOCATION: YOUR_GAR_LOCATION # TODO: update Artifact Registry location
+  SERVICE: YOUR_SERVICE_NAME # TODO: update Cloud Run service name
+  REGION: YOUR_SERVICE_REGION # TODO: update Cloud Run service region
+
+jobs:
+  deploy:
+    # Add "id-token" with the intended permissions for workload identity federation
+    permissions:
+      contents: "read"
```
Comment (sethvargo) [inline deployments/google-cloudrun-docker.yml:49]: We generally use single quotes in the YAML

**Review by verbanicm — COMMENTED** [review]
Comments addressed

File: deployments/google-cloudrun-docker.yml (hunk 4 above)
Comment (bharathkkb) [reply deployments/google-cloudrun-docker.yml]: Got it maybe we can join these two lines? While parsing the workflow, this did not click for me :) 

```
      # NOTE: If you already have a container image, these Docker steps can be omitted

      # BEGIN - Docker auth and build
```

**Pull request merged**